ci/pkg/ci/ci.go

```diff
@@ -0,0 +1,306 @@
+			return GetPodSpec(childYaml.(map[string]interface{}))
+		}
+	}
+	return yaml
+}
+
+func getImages(podSpec map[string]interface{}) []string {
+	images := make([]string, 0)
+	for _, field := range containerSpecFields {
+		containerField, ok := podSpec[field]
+		if !ok {
+			continue
+		}
+		containers := containerField.([]interface{})
+		for _, container := range containers {
+			images = append(images, container.(map[string]interface{})["image"].(string))
+		}
+	}
+	return images
+}
+
+// GetRepoTags returns the repotags from a tarball of a an image.
+func GetRepoTags(path string) ([]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	tarReader := tar.NewReader(f)
+	for {
+		header, err := tarReader.Next()
+
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		if header.Name != "manifest.json" {
+			continue
+		}
+		bytes, err := ioutil.ReadAll(tarReader)
+		if err != nil {
+			return nil, err
+		}
+		jsonBody := make([]interface{}, 0)
+		err = json.Unmarshal(bytes, &jsonBody)
+		if err != nil {
+			return nil, err
+		}
+		allRepoTags := make([]string, 0)
+		for _, imageDef := range jsonBody {
+			repoTags := imageDef.(map[string]interface{})["RepoTags"].([]interface{})
+			for _, tag := range repoTags {
+				allRepoTags = append(allRepoTags, tag.(string))
+			}
+		}
+		return allRepoTags, nil
+	}
+	return nil, nil
+}
+
+// SendResults sends the results to Insights
+func SendResults(trivyResults []byte, trivyVersion string, polarisVersion string, configurationObject Configuration, token string) error {
+	var b bytes.Buffer
+	w := multipart.NewWriter(&b)
+
+	var fw io.Writer
+	fw, err := w.CreateFormFile("trivy", "trivy.json")
+	if err != nil {
+		logrus.Warn("Unable to create form for Trivy")
+		return err
+	}
+	_, err = fw.Write(trivyResults)
+	if err != nil {
+		logrus.Warn("Unable to write contents for Trivy")
+		return err
+	}
+
+	fw, err = w.CreateFormFile("polaris", "polaris.json")
+	if err != nil {
+		logrus.Warn("Unable to create form for Polaris")
+		return err
+	}
+	r, err := os.Open(configurationObject.Options.TempFolder + "/polaris.json")
+	if err != nil {
+		logrus.Warn("Unable to open file for Polaris")
+		return err
+	}
+	defer r.Close()
+	_, err = io.Copy(fw, r)
+
+	if err != nil {
+		logrus.Warn("Unable to write contents for Polaris")
+		return err
+	}
+
+	w.Close()
+
+	masterHash, err := GetResultsFromCommand("git", "merge-base", "HEAD", "master")
+	if err != nil {
+		logrus.Warn("Unable to get GIT merge-base")
+		return err
+	}
+
+	currentHash, err := GetResultsFromCommand("git", "rev-parse", "HEAD")
+	if err != nil {
+		logrus.Warn("Unable to get GIT Hash")
+		return err
+	}
+
+	branchName, err := GetResultsFromCommand("git", "rev-parse", "--abbrev-ref", "HEAD")
+	if err != nil {
+		logrus.Warn("Unable to get GIT Branch Name")
+		return err
+	}
+
+	origin, err := GetResultsFromCommand("git", "remote", "get-url", "origin")
```

Comment:
might this contain an access key? should we remove pattern `://([^@]+@)`?

Reply:
I'm not positive, but better safe than sorry.